mtest/libmscore/text/tst_text.cpp

```diff
@@ -727,6 +728,41 @@ void TestText::testChineseBasicSupplemental()
       QCOMPARE(text->xmlText(), QString("你屠槪槪真軔")); // deleted the two chars in the middle
       }
 
+//---------------------------------------------------------
+///   testDropUnicodeAfterSMUFLwhenCursorSetToSymbol
+///     Tests dropping unicode after SMUFL as described in https://github.com/musescore/MuseScore/pull/3020#issuecomment-281932322
+///     When appeding text after a symbol, TextBlock needs to always (regardless of the state of the cursor) append a new TEXT TextFragment after the SYMBOL TextFragment.
+//---------------------------------------------------------
+
+void TestText::testDropUnicodeAfterSMUFLwhenCursorSetToSymbol()
+      {
+      Text* text = new Text(score);
+      text->initSubStyle(SubStyle::DYNAMICS);
+      text->setPlainText(QString(""));
+      text->layout();
+      text->startEdit(0, QPoint());
+
+      Symbol* symbolSMUFL = new Symbol(score); // create a new element, as Measure::drop() will eventually delete it
+      symbolSMUFL->setSym(SymId::noteheadWhole);
+
+      DropData dropSMUFL;
+      dropSMUFL.element = symbolSMUFL;
+      text->drop(dropSMUFL);
+
+      // the bug happened when cursor is in symbol mode
+      CharFormat* cf = text->cursor()->format();
+      cf->setType(CharFormatType::SYMBOL);
```

Comment:
This is a hack to force the cursor into symbol mode.  I would have liked to replicate all the exact steps I would perform with the GUI, but I couldn't figure out how to write those exactly as a mtest.  But the essential problem causing the bug was if the cursor was already in SYMBOL format before dropping Unicode.  So I forcibly set that here.